task_2/task_2.py

```diff
@@ -0,0 +1,430 @@
+
+    # 11 Store the data to csv files in corresponding folders. Name of file should be composed using
+    #     the following pattern: max_age_{}_avg_registered_{}_ popular_id_{}.csv
+    #     (max age, average number of years of registration, the most common id.name)
+    #     Log paths of created files.
+    fill_data_in_folder_structure(users_dict_list)
+
+    #12. Remove the data before 1960-th
+    remove_data_before_decade(users_dict_list, Var.min_decade)
+
+    #13. Log your full folder structure. Make tabulation to show nesting.
+    # Add a flag showing whether an item is a file or a folder.
+    draw_dir_structure(Var.output_folder, 0)
+
+    logger.info('-------- end script ---------')
+
+    # 14 Archive the destination folder.
+    arj_dir(Var.output_folder, Var.output_arj_name)
+
+
+def save_listdict_as_csv(paht_file, users_dict_list):
+    with open(paht_file, mode='w', newline='', encoding="utf-8") as file:
+        writer = csv.DictWriter(file, fieldnames=dict(users_dict_list[0]).keys())
+        writer.writeheader()
+        writer.writerows([data for data in users_dict_list])
+
+
+# 4.  Read the file and filter data by gender or number of rows in case the parameters were passed
+def filter_data(path_csv_file, gender, row_qantity):
+    with open(path_csv_file, 'r') as file:
+        data = file.readlines()
+
+    users_dict_list = [user for user in csv.DictReader(data[1:], fieldnames=data[0].split(','))]
+    if gender:
+        return [user for user in users_dict_list if user['gender'] == gender]
+    elif row_qantity:
+        return [user for user in users_dict_list[0:row_qantity]]
+    else:
+        return users_dict_list
+
+
+# '5.  Add fields to the file:
+#     - global_index (row number in csv file)
+#     - current_time (time of a user based on their timezone)
+#     - change the content in the field name.title using following rule:
+#           Mrs à missis; Ms à miss; Mr à mister; Madame à mademoiselle; other values should remain the same.
+#     - Convert dob.date to the format "month/day/year”
+#     - Convert register.date to the format "month-day-year, hours:minutes:second"
+# '
+def restruct_data(users_list_dict):
+
+    logger = logging.getLogger("restruct_data")
+    edited_data = []
+    for i, data_row in enumerate(users_list_dict):
+        data_row['global_index'] = i+1
+        hours, minutes = map(int, data_row['location.timezone.offset'].split(':'))
+        data_row['current_time'] = datetime.datetime.now(
+            datetime.timezone(datetime.timedelta(hours=hours, minutes=minutes))).strftime('%H:%M')
+
+        title_mapping = {
+            'Mrs': 'missis',
+            'Ms': 'miss',
+            'Mr': 'mister',
+            'Madame': 'mademoiselle'
+        }
+        data_row['name.title'] = title_mapping.get(data_row['name.title'], 'undefined')
+        if data_row['name.title'] == 'undefined':
+            logger.warning(f'User with global_index {i} - name.title = undefined!')
```

Comment:
**other values should remain the same.**
and you change them to undefined. fix it